### DISCUSSION TITLE
openEO

### DISCUSSION THREAD

**Discussion by mhungen** [microsoft/PlanetaryComputer#310] [Q&A]
Hallo there, planetarycomputer team!

Are there any future plans to offer an [openEO](https://openeo.org/) API? 
I think that could be a great addition to the platform and would make it even easier to enter the world of EO data processing.

regards,
Martin

**Comment by TomAugspurger** [comment]
Hi Martin. I don't have any experience with deploying or maintaining an openEO instance. I'd be curious how far someone could get with an openEO deployment that uses the Planetary Computer APIs and data, without necessarily being part of the Planetary Computer itself.

**Reply by m-mohr** [reply]
Hey @TomAugspurger, feel free to reach out to the openEO PSC (which I'm part of) if you are interested to discuss: openeo.psc@uni-muenster.de

Especially with the background of the security-based deprecation on the Hub, it might be worthwhile to think about a more security-restricted API alternative, if you still consider processing as a service for users. I had similar throughts when reading about the deprecation this morning, see also https://github.com/microsoft/PlanetaryComputer/discussions/347#discussioncomment-9519717

Generally speaking, an openEO API should be able to load from MS PC via the load_stac process, but there will be the data loading bottleneck, so having this as close to the data as possible is surely an advantage over having the openEO implementation hosted externally.

**Reply by TomAugspurger** [reply]
Thanks Matthias. I do want to explore openEO more when I have a chance.

> so having this as close to the data as possible is surely an advantage over having the openEO implementation hosted externally.

Indeed. The nice thing about the Planetary Computer architecture is that anyone is able to get access to the data with resources deployed in their own Azure subscription. They just need to be in the same region as the data (typically West Europe, but the Collection metadata includes information on which region the data is stored in).
